apollo-router/src/spec/query/tests.rs

```diff
@@ -522,9 +522,9 @@ fn reformat_response_data_best_effort() {
                         "baz": "2",
                     },
                     "array": [
-                        {},
+                        {"bar": null, "baz": "3"},
                         null,
-                        {}
+                        {"bar": "5", "baz": null}
                     ],
```

Comment:
I don’t know:
* How did response formatting end up emitting empty maps here, before this PR?
* What in this PR changed the output for this case?

… but the new output seems to me to be more correct. @Geal what do you think?

Reply:
@Geal we went through it while working on https://github.com/apollographql/router/pull/3718

IIRC I had a change that would output this, and we decided to not fix it that way, but I don't remember exactly why.
Can we walk through it again together tomorrow? gotta be 100% sure it's ok